Sources/SwiftFormat/PrettyPrint/WhitespaceLinter.swift

```diff
@@ -339,20 +340,15 @@ public class WhitespaceLinter {
     startingAt offset: Int,
     in data: [UTF8.CodeUnit]
   ) -> ArraySlice<UTF8.CodeUnit> {
-    func isWhitespace(_ char: UTF8.CodeUnit) -> Bool {
-      switch char {
-      case UInt8(ascii: " "), UInt8(ascii: "\n"), UInt8(ascii: "\t"), UInt8(ascii: "\r"), /*VT*/ 0x0B, /*FF*/ 0x0C:
-        return true
-      default:
-        return false
-      }
-    }
-    guard
-      let whitespaceEnd = data[offset...].firstIndex(where: { !isWhitespace($0) })
-    else {
-      return data[offset..<data.endIndex]
+    // To ensure consistency with PrettyPrinter, the data is converted to a String
+    // before processing whitespace.
+    let substring = String(decoding: data[offset...], as: UTF8.self)
+    var stringIndex = substring.startIndex
+    while stringIndex < substring.endIndex, substring[stringIndex].isWhitespace {
+      substring.formIndex(after: &stringIndex)
     }
-    return data[offset..<whitespaceEnd]
+    let utf8Count = substring.utf8.distance(from: substring.startIndex, to: stringIndex)
+    return data[offset..<offset + utf8Count]
```

Comment:
I modified the logic to determine whitespace to match the one used in [PrettyPrinter](https://github.com/swiftlang/swift-format/blob/main/Sources/SwiftFormat/PrettyPrint/Comment.swift#L17-L32).
However... after looking at https://github.com/swiftlang/swift-format/pull/915 and running performance tests, I found that it became significantly slower 🫠
```
before: 50321785572
after: 4392453228823
```
I think I'll need to look for an alternative approach.